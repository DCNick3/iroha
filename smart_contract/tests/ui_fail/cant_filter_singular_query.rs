use iroha_smart_contract::{
    data_model::query::predicate::{string::StringPredicate, value::QueryOutputPredicate},
    prelude::*,
};

fn main() {
    FindPermissionSchema
        .filter(QueryOutputPredicate::Identifiable(
            StringPredicate::starts_with("xor_"),
        ))
        .execute()
}
