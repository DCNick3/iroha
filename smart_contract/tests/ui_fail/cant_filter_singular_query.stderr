error[E0599]: the method `filter` exists for struct `FindPermissionSchema`, but its trait bounds were not satisfied
 --> tests/ui_fail/cant_filter_singular_query.rs:8:10
  |
7 | /     FindPermissionSchema
8 | |         .filter(QueryOutputPredicate::Identifiable(
  | |         -^^^^^^ method cannot be called on `FindPermissionSchema` due to unsatisfied trait bounds
  | |_________|
  |
  |
 ::: $WORKSPACE/data_model/src/query/mod.rs
  |
  | /     queries! {
  | |         /// Finds all registered permission tokens
  | |         #[derive(Copy, Display)]
  | |         #[ffi_type]
... |
  | |         }
  | |     }
  | |_____- doesn't satisfy `_: ExecuteIterableQueryOnHost`, `_: IterableQuery` or `_: Iterator`
  |
  = note: the following trait bounds were not satisfied:
          `iroha_smart_contract::prelude::FindPermissionSchema: IterableQuery`
          which is required by `iroha_smart_contract::prelude::FindPermissionSchema: iroha_smart_contract::ExecuteIterableQueryOnHost`
          `iroha_smart_contract::prelude::FindPermissionSchema: Iterator`
          which is required by `&mut iroha_smart_contract::prelude::FindPermissionSchema: Iterator`
